Source/Core/EllipsoidGeometry.js

```diff
@@ -206,6 +206,10 @@ define([
         var cosTheta = new Array(slicePartitions);
         var sinTheta = new Array(slicePartitions);
 
+        if ((radii.x === 0) || (radii.y === 0) || (radii.z === 0)) {
```

Comment:
Yeah, that's why we want to move this check to the top of the function.  So the `Ellipsoid` will never be created.